### PR TITLE
update functions-and-operator and generated-columns

### PR DESCRIPTION
**Pull request by niezefeng**
### What is changed, added or deleted? (Required)
Update functions-and-operator documents and json-functions documents.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3318, https://github.com/pingcap/docs-cn/pull/3295
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
PTAL @Reminiscent

File: generated-columns.md [hunk 1]
```diff
@@ -26,65 +38,118 @@ CREATE TABLE person (
...
+| IndexReader_6          | 10.00   | root      |                         | index:IndexRangeScan_5                      |
+| └─IndexRangeScan_5     | 10.00   | cop[tikv] | table:t, index:idx_b(b) | range:[3,3], keep order:false, stats:pseudo |
++------------------------+---------+-----------+-------------------------+---------------------------------------------+
+2 rows in set (0.01 sec)
 ```
 
-## Limitations
+> **Note:**
+>
+> Only when the expression type and the generated column type are strictly equal, the replacement will be performed.
```
Comment (ran-huang) [inline generated-columns.md:142]:
```suggestion
> Only when the expression type and the generated column type are strictly equal, the replacement is performed.
```

File: generated-columns.md [hunk 2]
```diff
@@ -26,65 +38,118 @@ CREATE TABLE person (
...
+>
+> In the above example, the column type of `a` is int and the column type of `a+1` is bigint. If the type of the generated column is set to int, the replacement will not occur.
+>
+> For type conversion rules, see [Type Conversion of Expression Evaluation] (/functions-and-operators/type-conversion-in-expression-evaluation.md).
+
+## Limitations of Generated columns
```
Comment (ran-huang) [inline generated-columns.md:148]:
```suggestion
## Limitations
```

File: generated-columns.md [hunk 3]
```diff
@@ -26,65 +38,118 @@ CREATE TABLE person (
...
 
 The current limitations of JSON and generated columns are as follows:
 
-- You cannot add the generated column in the storage type of `STORED` through `ALTER TABLE`.
+- You cannot add the generated column through `ALTER TABLE`.
 - You can neither convert a generated stored column to a normal column through the `ALTER TABLE` statement nor convert a normal column to a generated stored column.
```
Comment (ran-huang) [inline generated-columns.md:153]:
```suggestion
- You can neither convert a stored generated column to a normal column through the `ALTER TABLE` statement nor convert a normal column to a stored generated column.
```

File: generated-columns.md [hunk 4]
```diff
@@ -26,65 +38,118 @@ CREATE TABLE person (
...
-- You cannot modify the **expression** of a generated stored column through the `ALTER TABLE` statement.
-- Not all [JSON functions](/functions-and-operators/json-functions.md) are supported.
+- Not all [JSON functions](/functions-and-operators/json-functions.md) are supported;
```
Comment (ran-huang) [inline generated-columns.md:154]:
```suggestion
- You cannot modify the expression of a stored generated column through the `ALTER TABLE` statement.
- Not all [JSON functions](/functions-and-operators/json-functions.md) are supported;
```

File: generated-columns.md [hunk 5]
```diff
@@ -26,65 +38,119 @@ CREATE TABLE person (
...
     id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
     name VARCHAR(255) NOT NULL,
     address_info JSON,
-    city VARCHAR(64) AS (JSON_UNQUOTE(JSON_EXTRACT(address_info, '$.city'))) STORED NOT NULL,
+    city VARCHAR(64) AS (JSON_UNQUOTE(JSON_EXTRACT(address_info, '$.city'))) NOT NULL,
     KEY (city)
 );
 ```
 
+## Validation of Generated columns
```
Comment (ran-huang) [inline generated-columns.md:98]:
```suggestion
## Validation of generated columns
```

File: generated-columns.md (hunk 4 above)
Comment (ran-huang) [reply generated-columns.md:155]: @niezefeng ``不能通过 `ALTER TABLE` 修改存储生成列的生成列表达式`` Is this limitation no longer applicable?

File: generated-columns.md [hunk 6]
```diff
@@ -26,65 +38,120 @@ CREATE TABLE person (
...
+
 ## Limitations
 
 The current limitations of JSON and generated columns are as follows:
 
-- You cannot add the generated column in the storage type of `STORED` through `ALTER TABLE`.
-- You can neither convert a generated stored column to a normal column through the `ALTER TABLE` statement nor convert a normal column to a generated stored column.
-- You cannot modify the **expression** of a generated stored column through the `ALTER TABLE` statement.
-- Not all [JSON functions](/functions-and-operators/json-functions.md) are supported.
+- You cannot add the generated column through `ALTER TABLE`.
```
Comment (wjhuang2016) [inline generated-columns.md:152]: You cannot add the stored generated column through `ALTER TABLE`.

File: generated-columns.md [hunk 7]
```diff
@@ -26,65 +38,120 @@ CREATE TABLE person (
...
+- You can neither convert a stored generated column to a normal column through the `ALTER TABLE` statement nor convert a normal column to a stored generated column.
+- You cannot modify the expression of a stored generated column through the `ALTER TABLE` statement.
```
Comment (wjhuang2016) [inline generated-columns.md:154]: Duplicated with L155

**Comment by ran-huang** [comment]
@wjhuang2016 Comments addressed. PTAL again. Thanks~

**Pull request merged**